lib/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.3)
+project(Libraries)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+
+set(SOURCE_FILES main.cpp)
+set(LIBRARY_FILES lib/reader.h lib/reader.cpp)
```

Comment:
I don't have this line `set(LIBRARY_FILES lib/reader.h lib/reader.cpp)` in either of my CMakeLists, maybe that is the problem. Also it might give you an error if there is an error in your library program too. You will have to build the library too in order for it to show up in the debugging folder.